src/prog_algs/state_estimators/unscented_kalman_filter.py

```diff
@@ -74,8 +71,26 @@ def state_transition(x, dt):
         num_measurements = model.n_outputs
         points = kalman.MerweScaledSigmaPoints(num_states, alpha=self.parameters['alpha'], beta=self.parameters['beta'], kappa=self.parameters['kappa'])
         self.filter = kalman.UnscentedKalmanFilter(num_states, num_measurements, self.parameters['dt'], measure, state_transition, points)
-        self.filter.x = array(list(x0.values())).ravel()
-        self.filter.P = self.parameters['Q'] / 10
+        
+        parameter_R_bool = 'R' not in self.parameters
```

Comment:
Why is this saved as a different parameter- I think you can put it in the conditional below. 

Caching only helps when its before an array, or is executed a large number of times

Reply:
Another thought, can we put this comparison after the if/elif control structure and just set R to diag([1.0e-3 for i in range(len(measure(self.filter.x)))])? That way we don't need to define it twice